desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,264 @@
+        requiring more memory. A ``jac_chunk_size`` of 1 corresponds to the least
+        memory intensive, but slowest method of calculating the Jacobian.
+        If None, it will use the largest size i.e ``obj.dim_x``.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε: "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        rho=1.0,
+        alpha=0.0,
+        *,
+        knots_per_transit=100,
+        num_transit=10,
+        num_quad=32,
+        num_pitch=50,
+        batch=True,
+        num_well=None,
+        name="Effective ripple",
+        jac_chunk_size=None,
+    ):
+        if target is None and bounds is None:
+            target = 0.0
+
+        rho, alpha = np.atleast_1d(rho, alpha)
+        self._dim_f = rho.size
+        self._keys_1dr = [
+            "iota",
+            "iota_r",
+            "<|grad(rho)|>",
+            "min_tz |B|",
+            "max_tz |B|",
+            "R0",  # TODO: GitHub PR #1094
+        ]
+        self._constants = {
+            "quad_weights": 1,
+            "rho": rho,
+            "alpha": alpha,
+            "zeta": np.linspace(
+                0, 2 * np.pi * num_transit, knots_per_transit * num_transit
+            ),
+            "quad": chebgauss2(num_quad),
+        }
+        self._hyperparameters = {
+            "num_pitch": num_pitch,
+            "batch": batch,
+            "num_well": num_well,
+        }
+
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+            jac_chunk_size=jac_chunk_size,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+        self._grid_1dr = LinearGrid(
+            rho=self._constants["rho"], M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, sym=eq.sym
+        )
+        self._target, self._bounds = _parse_callable_target_bounds(
+            self._target, self._bounds, self._constants["rho"]
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        self._constants["transforms_1dr"] = get_transforms(
+            self._keys_1dr, eq, self._grid_1dr
+        )
+        self._constants["profiles"] = get_profiles(
```

Comment:
When you define profiles for the effective ripple, why does it take in grid_1dr?
Shouldn't the profiles object for "effective ripple" take in the  "grid" from eq._get_rtg_grid ?

Reply:
The flux surface of each grid point is the same so it doesn't matter since the profile is a flux function.

The get profiles method just checks if the grid has a node at magnetic axis.